### DISCUSSION TITLE
How to edit files?

### DISCUSSION THREAD

**Discussion by OSA413** [WerWolv/ImHex#535] [Get Help] [answered]
Like in some other hex editors (GHex or Frhed), I can just open a file, place my cursor in the hex view and start typing numbers, I see the results of the changes in the hex view and pressing Ctrl + S saves the file on the disk. I couldn't find a way of doing similar things here (I tried to search for tutorials on YouTube but found none, the documentation is empty) so I'm posting this discussion.

**Comment by WerWolv** [comment, marked as answer]
Very easy, double click a byte and it goes into editing mode where you can start typing. It will automatically go to the next byte once you've entered enough characters

**Reply by OSA413** [reply]
When I double click, nothing happens

**Reply by WerWolv** [reply]
![imhex_9GpfEmdTzu](https://user-images.githubusercontent.com/10835354/173663923-e1200e1a-414a-4008-999d-e451d64c9483.gif)
Works perfectly fine here. Are you using the latest version? Or are you possibly trying to edit a file you don't have write access to?

**Reply by OSA413** [reply]
Oh, turns out that the window didn't refresh so I couldn't see that cursor allowed me to edit the bytes.

**Reply by OSA413** [reply]
I have a video that I recorded before I found it out, I need to convert it

**Reply by OSA413** [reply]
...no, it doesn't allow to edit again or either didn't refresh

**Reply by WerWolv** [reply]
It does update and work fine here.

Can you give some more context? What version are you using? What OS are you on? etc

**Reply by OSA413** [reply]
Windows 10, installed using MSI installer, UI scale is 1.5x, font is Liberation Mono Regular, 14pt.

**Reply by OSA413** [reply]
https://youtu.be/LUy5gjkA6Aw

**Reply by OSA413** [reply]
The version is 1.18.2

**Reply by WerWolv** [reply]
That video is set to private, I can't watch it

**Reply by OSA413** [reply]
Ooops, should work now.

**Reply by WerWolv** [reply]
Stupid question but are you double clicking quick enough? It won't work if you do it too slow.
Other than that, can you try with a different file maybe? Or does 1.0x scaling work maybe?

**Reply by OSA413** [reply]
Lol, it was the click speed, I was using touchpad and couldn't tap fast enough with one finger to activate the edit mode. Can you please increase the time window for the double click?

**Reply by WerWolv** [reply]
I don't think I can, no. It uses native double click detection as done by the OS.
I can do it fine with the touch pad of my laptop though, have you tried just double tapping the touch pad instead of actually fully clicking down the button?

**Reply by OSA413** [reply]
For some reason it doesn't activate the edit mode if I tap on the same exact place on the touch pad, but if I change the place for the second tap or use two fingers (since they have some distance between them) it works just fine, I don't even need to tap very fast, just at usual speed. Though this thing/bug doesn't appear when I use File Explorer or other things with double tap.

**Reply by OSA413** [reply]
I think I should try to report the error in the original repository of the GUI you are using for the tool.

**Reply by OSA413** [reply]
Turn out that the issue was present in 1.83 WIP of ImGUI and should be fixed in 1.88 WIP (according to the compiled demo of ImGUI)
https://www.dearimgui.org/binaries/

**Comment by DomHeadroom** [comment]
> Very easy, double click a byte and it goes into editing mode where you can start typing. It will automatically go to the next byte once you've entered enough characters

Is there a way to remove a selected part, from a file, using imHex? Because i have tried pressing the del key and the backspace but nothing happens, in the context menu there is no option either!

**Reply by WerWolv** [reply]
There will be with the next version. `Edit -> Remove...`

**Reply by DomHeadroom** [reply]
> There will be with the next version. `Edit -> Remove...`

Ok, thank you for the answer!

**Comment by rabbyzero** [comment]
Is there a way to start editing without  mouse/touchpad click. Most of actions can be done with keyboard, while editing requires my hand to leave keyboard.

**Reply by paxcut** [reply]
in the hex editor you need the mouse click to start editing but you can use Ctrl-c and ctrl-v to copy and paste even if you are not editing just as long as the data is highlighted which you can obtain using shift+arrow keys
